org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/JavacUtils.java

```diff
@@ -385,7 +385,7 @@ private static Collection<File> classpathEntriesToFiles(JavaProject project, Pre
 								for (IClasspathEntry transitiveEntry : resolved ) {
 									if (transitiveEntry.getEntryKind() == IClasspathEntry.CPE_SOURCE) {
 										IPath outputLocation = transitiveEntry.getOutputLocation();
-										if (outputLocation != null && select.test(transitiveEntry)) {
+										if (outputLocation != null && !transitiveEntry.isTest()) {
```

Comment:
The `select` condition needs to be a parameter, it may take different values depending on the case. So instead of hardcoding the condition here, I think it should instead be tweaked in the parameter of the callers.
Concretely here, what is missing? Is it an entry in the sourcepath or in the classpath? If the former, tweaking line 268 seems better, if the former then line 303 would be a more accurate location to change,